condorpy/workflow.py

```diff
@@ -156,6 +171,97 @@ def _update_status(self, sub_job_num=None):
 
         return key
 
+    def _update_statuses(self, sub_job_num=None):
+        """
+        Update statuses of jobs nodes in workflow.
+        """
+        # initialize status dictionary
+        status_dict = dict()
+
+        for val in CONDOR_JOB_STATUSES.itervalues():
+            status_dict[val] = 0
+
+        for node in self.node_set:
+            job = node.job
+            try:
+                job_status = job.status
+                status_dict[job_status] += 1
+            except (KeyError, HTCondorError):
+                status_dict['Unexpanded'] += 1
+
+        return status_dict
+
+    def update_node_ids(self, sub_job_num=None):
```

Comment:
New method that associates nodes with the job running on condor (cluster_id). Queries for jobs associated with the DAG and uses combination of executable (cmd) and arguments (args) to match the Node job with the job in condor_q/condor_history to get the cluster_id for the job. Once cluster_id is assigned, Job.status works as usual.